### DISCUSSION TITLE
Posible to send binary file and data through websocket?

### DISCUSSION THREAD

**Discussion by Wyzix33** [uNetworking/uWebSockets.js#738] [Q&A]
I'm having some small tachometer (esm) files (20-30 kb) and would like to send them through to client using `ws.send` but together with some other data (json).
I've succeeded reading the file as base64 and sending it as part of the json object, but would like to send it as arraybuffer as that is what i need on the client, is this possible :
```
const buff = await readFile(saferesolve('./privat/upload', file));
ws.send({ type: 'theFile', someOtherProps: 'some value', AB: buff }); 
```
or something like that...
With this code I'm getting invalid array buffer on the client, it's like it's Uint8Array having the value of the array buffer or something like that :) I don't have much experience with buffers..
I'm getting 
Uint8Array(133175)} and buffer: ArrayBuffer(133209)
 and it should be ArrayBuffer(133175)
so the ArrayBuffer from nodejs gets to be Uint8Array on browser ...
Any help with this?
Thanks

**Comment by e3dio** [comment]
You can't send buffer/binary inside JSON string unless you convert to string format with base64, it will use 33% more space, then decode base64 on client. Better send buffer data as separate message in binary format `ws.send(buf, true)`. You could add other fields to your binary buffer message or just send it after JSON message
